src/configuration.cpp

```diff
@@ -0,0 +1,702 @@
+                   .long_description(unindent(R"(
+                        'ssl_verify' can be either an empty string (regular SSL verification),
+                        the string "<false>" to indicate no SSL verification, or a path to
+                        a directory with cert files, or a cert file..)"))
+                   .set_post_build_hook(detail::ssl_verify_hook));
+
+        insert(Configurable("cacert_path", std::string(""))
+                   .group("Network")
+                   .set_env_var_name()
+                   .description("Path (file or directory) SSL certificate(s)")
+                   .long_description(unindent(R"(
+                        Path (file or directory) SSL certificate(s) to use whe
+                        'ssl_verify' in turned on but not set with path to certs.
+                        WARNING: overrides 'ssl_verify' if provided and 'ssl_verify'
+                        also contains a path to SSL certificates.)")));
+
+        insert(Configurable("ssl_no_revoke", &ctx.ssl_no_revoke)
+                   .group("Network")
+                   .set_env_var_name()
+                   .description("SSL certificate revocation checks")
+                   .long_description(unindent(R"(
+                        This option tells curl to disable certificate revocation checks.
+                        It's only working for 'Schannel' back-end.
+                        WARNING: this option loosens the SSL security, and by using this
+                        flag you ask for exactly that.)")));
+        // Solver
+        insert(Configurable("channel_priority", &ctx.channel_priority)
+                   .group("Solver")
+                   .set_env_var_name()
+                   .description("Define the channel priority ('strict', 'flexible' or 'disabled')")
+                   .long_description(unindent(R"(
+                    Accepts values of 'strict', 'flexible', and 'disabled'. The default
+                    value is 'flexible'. With strict channel priority, packages in lower
+                    priority channels are not considered if a package with the same name
+                    appears in a higher priority channel. With flexible channel priority,
+                    the solver may reach into lower priority channels to fulfill
+                    dependencies, rather than raising an unsatisfiable error. With channel
+                    priority disabled, package version takes precedence, and the
+                    configured priority of channels is used only to break ties. In
+                    previous versions of conda, this parameter was configured as either
+                    True or False. True is now an alias to 'flexible'.)")));
+
+        insert(Configurable("pinned_packages", &ctx.pinned_packages)
+                   .group("Solver")
+                   .set_env_var_name()
+                   .description("A list of package specs to pin for every environment resolution"));
```

Comment:
does this want a path to a file? or a list as cmd line options?

Reply:
This has the same type as the context value, which is `std::vector<std::string>>` that represent the specs.